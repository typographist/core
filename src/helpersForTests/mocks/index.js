const breakpointsModel = require('./breakpointsModel');
const userConfig = require('./userConfig');

module.exports = {
  breakpointsModel,
  userConfig,
};
