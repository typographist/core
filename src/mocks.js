// @flow

type Breakpoint = {|
  base: number | number[],
  lineHeight: number,
  name: string,
  ratio: number,
  root: number,
  value: string,
|};

export type Breakpoints = Breakpoint[];

export const breakpoints: Breakpoints = [
  {
    base: [16, 33],
    lineHeight: 1.5,
    name: 'default',
    ratio: 1.1342044031840226,
    root: 12,
    value: '0px',
  },
  {
    base: 17,
    lineHeight: 1.5,
    name: 'tablet',
    ratio: 1.1258335640240535,
    root: 13,
    value: '640px',
  },
  {
    base: 18,
    lineHeight: 1.7,
    name: 'desktop',
    ratio: 1.333,
    root: 16,
    value: '1024px',
  },
  {
    base: 20,
    lineHeight: 1.7,
    name: 'lgDesktop',
    ratio: 1.333,
    root: 17,
    value: '1200px',
  },
  {
    base: 22,
    lineHeight: 1.7,
    name: 'xlDesktop',
    ratio: 1.333,
    root: 19,
    value: '1600px',
  },
];

export const userConfig = {
  base: ['1em', '2em'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    value: '40em',
    base: '17px',
  },
  desktop: {
    value: '64em',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    value: '75em',
    base: '20px',
  },
  xlDesktop: {
    value: '100em',
    base: '22px',
  },
};
