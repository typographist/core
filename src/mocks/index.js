export { breakpoints, breakpointsModel } from './breakpoints';
export { userConfig, invalidUserConfig } from './user-configs';
