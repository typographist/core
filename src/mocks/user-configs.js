export const userConfig = {
  base: ['16px', '32px'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    breakpoint: '768px',
    base: '17px',
  },
  desktop: {
    breakpoint: '992px',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '1200px',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '1600px',
    base: '22px',
  },
};

export const invalidUserConfig = {
  base: ['1rem', '2em'],
  lineHeight: 1.5,
  ratio: '45 at 6',
  tablet: {
    base: '17px',
  },
  desktop: {
    breakpoint: '64rem',
    base: '18px',
    lineHeight: '1.7',
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '75rem',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '100rem',
  },
};
