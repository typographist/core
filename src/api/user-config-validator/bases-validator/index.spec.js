import { userConfig as config } from '../../../mocks';
import {
  getBases,
  hasPxOrEmWithException,
  isValidField,
  validateFields,
} from '.';

describe('bases validator', () => {
  describe('getBases', () => {
    it('returs all values of base from user config', () => {
      expect(getBases(config)).toEqual([
        ['16px', '33px'],
        '17px',
        '18px',
        '20px',
        '22px',
      ]);
    });
  });

  describe('hasPxOrEmWithException', () => {
    it('if the base value has pixels', () => {
      expect(hasPxOrEmWithException('12px')).toBe(true);
    });

    it('if the value contain string with em unit', () => {
      expect(hasPxOrEmWithException('12em')).toBe(true);
    });

    it('if base contain invalid value', () => {
      expect(hasPxOrEmWithException).toThrowErrorMatchingSnapshot();
    });
  });

  describe('isValidField', () => {
    it('if the base field is Array of strings contain px unit', () => {
      expect(isValidField(['78px', '28px'])).toBe(true);
    });

    it('if the base is Array of strings contain em unit', () => {
      expect(isValidField(['78em', '28em'])).toBe(true);
    });

    it('if the base is Array of strings contain em unit', () => {
      expect(isValidField(['78em', '28em'])).toBe(true);
    });

    it('will throw an exception', () => {
      expect(isValidField).toThrowErrorMatchingSnapshot();
    });
  });

  describe('validateFields', () => {
    it('all bases are valid', () => {
      expect(validateFields(config)).toBe(true);
    });
  });
});
