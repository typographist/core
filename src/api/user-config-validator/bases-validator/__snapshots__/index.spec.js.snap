// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bases validator hasPxOrEmWithException if base contain invalid value 1`] = `"Typographist >>> Check your user config. base:  must contain the values in pixels or ems. For example base: '6em' or base: '38px'."`;

exports[`bases validator isValidField will throw an exception 1`] = `"Typographist >>> Check your user config. base:  must must be a string or an array of strings. For example base: ['16px', '32px'] or base: '16px'."`;
